File: packages/frontend/src/types/TEventParticipation.ts
/*
  Hollybike Back-office web application
  Made by MacaronFR (Denis TURBIEZ) and enzoSoa (Enzo SOARES)
*/
import { TUserPartial } from "./TUserPartial.ts";
import { EEventRole } from "./EEventRole.ts";
import { TUserJourney } from "./TUserJourney.ts";

export interface TEventParticipation {
	user: TUserPartial,
	role: EEventRole,
	isImagePublic: boolean,
	joinedDateTime: Date,
	journey?: TUserJourney
}

File: packages/backend/src/main/kotlin/hollybike/api/types/event/participation/TEventParticipation.kt
package hollybike.api.types.event.participation

import hollybike.api.repository.EventParticipation
import hollybike.api.types.user.TUserPartial
import kotlinx.datetime.Instant
import kotlinx.serialization.Serializable

@Serializable
data class TEventParticipation(
	val user: TUserPartial,
	val role: EEventRole,
	val isImagesPublic: Boolean,
	val joinedDateTime: Instant,
	val journey: TUserJourney? = null
) {
	constructor(entity: EventParticipation, isBetterThan: Map<String, Double>?) : this(
		user = TUserPartial(entity.user),
		role = entity.role,
		isImagesPublic = entity.isImagesPublic,
		joinedDateTime = entity.joinedDateTime,
		journey = entity.journey?.let { TUserJourney(it, isBetterThan) }
	)
}
